src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/CSharp/TypeAccessibilityCodeActionProviderTest.cs

```diff
@@ -0,0 +1,396 @@
+using Microsoft.CodeAnalysis.Text;
+using Moq;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Xunit;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions
+{
+    public class TypeAccessibilityCodeActionProviderTest : LanguageServerTestBase
+    {
+        [Fact]
+        public async Task Handle_MissingDiagnostics_ReturnsEmpty()
+        {
+            // Arrange
+            var documentPath = "c:/Test.razor";
+            var contents = "";
+            var request = new CodeActionParams()
+            {
+                TextDocument = new TextDocumentIdentifier(new Uri(documentPath)),
+                Range = new Range(),
+                Context = new CodeActionContext()
+                {
+                    Diagnostics = null
+                }
+            };
+
+            var location = new SourceLocation(0, -1, -1);
+            var context = CreateRazorCodeActionContext(request, location, documentPath, contents, new SourceSpan(0, 0));
+            context.CodeDocument.SetFileKind(FileKinds.Legacy);
+
+            var provider = new TypeAccessibilityCodeActionProvider();
+            var csharpCodeActions = new[] {
+                new RazorCodeAction()
+                {
+                    Title = "System.Net.Dns"
+                }
+            };
+
+            // Act
+            var results = await provider.ProvideAsync(context, csharpCodeActions, default);
+
+            // Assert
+            Assert.Empty(results);
+        }
+
+        [Fact]
+        public async Task Handle_InvalidDiagnostics_ReturnsEmpty()
+        {
+            // Arrange
+            var documentPath = "c:/Test.razor";
+            var contents = "";
+            var request = new CodeActionParams()
+            {
+                TextDocument = new TextDocumentIdentifier(new Uri(documentPath)),
+                Range = new Range(),
+                Context = new CodeActionContext()
+                {
+                    Diagnostics = new Container<Diagnostic>(
+                        new Diagnostic()
+                        {
+                            // Invalid as Error is expected
+                            Severity = DiagnosticSeverity.Warning,
+                            Code = new DiagnosticCode("CS0246")
+                        },
+                        new Diagnostic()
+                        {
+                            // Invalid as CS error code is expected
+                            Severity = DiagnosticSeverity.Error,
+                            Code = new DiagnosticCode(0246)
+                        },
+                        new Diagnostic()
+                        {
+                            // Invalid as CS0246 or CS0103 is expected
+                            Severity = DiagnosticSeverity.Error,
+                            Code = new DiagnosticCode("CS0183")
```

Comment:
Comment says CS0103 but code says CS0183, I would assume they should match? Sees like it the code is wrong the test should have failed (if the comment is at fault it makes sense).

Reply:
CS0103 or CS0246 are the expected codes. CS0183 is "wrong", and hence the test fails by returning 0 results. This test case is just to ensure we don't incorrectly interpret unrelated/invalid diagnostic codes as valid for this code action provider.